src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -120,22 +147,36 @@ public TermQueryBuilder boost(float boost) {
     }
 
     /**
-     * Sets the query name for the filter that can be used when searching for matched_filters per hit.
+     * Gets the boost for this query.
+     */
+    public float boost() {
+        return this.boost;
+    }
+
+    /**
+     * Sets the query name for the query.
      */
     public TermQueryBuilder queryName(String queryName) {
         this.queryName = queryName;
         return this;
     }
 
+    /**
+     * Gets the query name for the query.
+     */
+    public String queryName() {
```

Comment:
I'd probably try doing this for all queries in #10776, started looking at this on separate branch already. There's a few more things to clean here, so I'd probably not mix it but revisit this query again in separate PR.